src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -67,61 +58,44 @@ export default function CodeEditor(props) {
     });
 
     const data = await response.json();
-    console.log("data: ", data);
     const errorInfo = checkError(data);
-    if (errorInfo != null) {
-      props.onCompile(errorInfo)
+    if (errorInfo.length > 0) {
+      props.onCompile(errorInfo);
       return;
     }
-    
-    let contractName = "";
+
     const output = Object.keys(data.contracts["test.sol"]).map((key) => {
-      contractName = key;
       return {
         name: key,
         abi: data.contracts["test.sol"][key].abi,
         bytecode: data.contracts["test.sol"][key].evm.bytecode.object,
       };
     });
     props.onCompile(output);
-    //console.log("output: ", output)
-    
-    setCompilerData(output)
+    setCompilerData(output);
   }
 
   async function deploy(compilerData) {
-    console.log("data: ", compilerData);
-
-    var web3 = new Web3(window.ethereum);
-
-    const response = await fetch("http://localhost:3001/deploy", {
-      headers: {
-        "Content-Type": "application/json",
-      },
-      method: "POST",
-      body: JSON.stringify({ value: compilerData,
-                             address: address, }),
-    });
-
-    const { deploy } = await response.json();
-    console.log("data: ", deploy);
-
-    let gas = 1000;
-    console.log(gas)
-    let gasPrice = 1000;
-    console.log(gasPrice)
-    console.log(address)
-    let transactionObject ={
-      gas: gas,
-      gasPrice: gasPrice,
-      data: deploy[0], // TODO: Make a loop
-      from: address
+    for (let i = 0; i < compilerData.length; i++) {
+      const { bytecode } = compilerData[i];
+      let transactionObject = {
+        data: "0x" + bytecode,
+        from: address,
+      };
+      const gas = await window.ethereum.request({
+        method: "eth_estimateGas",
+        params: [transactionObject],
+      });
+      transactionObject.gas = gas;
+
+      const res = await window.ethereum.request({
+        method: "eth_sendTransaction",
+        params: [transactionObject],
+      });
```

Comment:
Check out https://eth.wiki/json-rpc/API#eth_sendtransaction for the docs on other rpc interactions. Seems like we can use this instead of web3 entirely. 

I'm really interested in possibly creating a library to abstract this stuff because the current libraries available are not great. ( maybe post grad project if anyone's interested )